tests/integration/test_s3.py

```diff
@@ -889,6 +889,22 @@ def test_upload_big_file(self):
         # clean up
         self._delete_bucket(bucket_name, [key1, key2])
 
+    def test_s3_put_more_than_1000_items(self):
```

Comment:
While this test case is legitimate, I think the original issue reported was related to the `list_object` API, which only returned 1000 entries at a time.

```
val s = s3Client.listObjects(srcBucketName).getObjectSummaries.asScala.toList.size
```

Could we extend this test to also fetch the entire list of 1010 objects - 1000 in a first iteration via `list_objects`, and then the remaining 10 items by using the `NextMarker` returned from `list_objects`.